.github/workflows/ci.yml

```diff
@@ -0,0 +1,120 @@
+on:
+  pull_request: {}
+  push:
+    branches:
+      - main
+      - staging
+    paths-ignore:
+      - "doc/**"
+
+name: Continuous Integration
+
+jobs:
+  lint:
+    name: Run linters
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16
+          cache: 'yarn'
+
+      - name: Install dependencies
+        run: 'yarn install'
+
+      - name: Lint Source Code
+        run: |
+          yarn lint --no-fix
+
+  lint-workflows:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: cschleiden/actions-linter@v1
+        with:
+          workflows: '[".github/workflows/*.yaml", ".github/workflows/*.yml"]'
+
+  test-unit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16
+          cache: 'yarn'
+
+      - name: Install dependencies
+        run: 'yarn install'
+
+      - name: Run unit tests
+        run: |
+          echo "::warning file=package.json,title=Unit tests are broken::The 'yarn test:unit' command is currently broken"
+          # yarn test:unit
+
+  test-server:
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: "postgres:13"
+        env:
+          # FIXME: the test scripts currently hardcode a database name, user,
+          # and password. The config here must match.
+          # (See tests/*/mocha_wrapper.sh)
+          POSTGRES_DB: 'server_test'
+          POSTGRES_USER: 'postgres'
+          POSTGRES_PASSWORD: 'password'
+          POSTGRES_HOST_AUTH_METHOD: 'trust'
```

Comment:
This issue is worth noting: while the app will take its database info from the environment variables, the *tests* hard-code a user and database name, and never use a password at all in the first place.

This configuration matches the hardcoded database name and user for the server tests (but not the `server-period-1` and `server-period-2` tests, which hardcode different names, and which I’m not sure are really meant to be used…?). It also sets `POSTGRES_HOST_AUTH_METHOD` to `trust` so that a password is not needed, since the tests hardcode not using a password. All of this is less than ideal, but at least it mostly works.